### PR TITLE
Fix websocket close event propagation on unclean disconnect

### LINKED ISSUE #7012: Unclean websocket disconnection is not propagated to endpoints
## Environment Details
* Helidon Version: 3.2.1
* Both Helidon SE and Helidon MP
* JDK version: Amazon Corretto 17
* OS: Alpine 3.18 and MacOS

----------

## Problem Description
When a client disconnects without calling close on the websocket session the server endpoint class is not notified about the disconnection.
If the developer didn't set the idle timeout the stale session will remain "active" (the basic and async remotes will return true to "isOpen()" function calls).
The internal executor service is shutdown when disconnecting resulting in rejection exception when the application tries to use the now closed session.

Original Exception:
```
java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@b88be9c[Not completed, task = java.util.concurrent.Executors$RunnableAdapter@3b5d5e5b[Wrapped task = org.glassfish.tyrus.core.TyrusSession$IdleTimeoutCommand@75480a9b]] rejected from java.util.concurrent.ScheduledThreadPoolExecutor@456042d[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 7]
	at java.base/java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2065)
	at java.base/java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:833)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:340)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.schedule(ScheduledThreadPoolExecutor.java:562)
	at org.glassfish.tyrus.core.TyrusSession.restartIdleTimeoutExecutor(TyrusSession.java:489)
	at org.glassfish.tyrus.core.TyrusRemoteEndpoint$Basic.sendText(TyrusRemoteEndpoint.java:85)
	...
```

## Steps to reproduce
Run a simple server with an echo endpoint thats sends a message each seconds, connect a client to it and then kill the client without explicitly closing the client before killing it.

### PR DESCRIPTION
**Pull request by asafalima**
Fixes #7012

### CODE REVIEW

**Comment by asafalima** [comment]
I will rebase my branch after your fix for #7002

**Comment by asafalima** [comment]
@spericas I rebased the branch from the upstream after you merged your fix for #7002, I would appreciate your review when you get the chance.
I've found another critical bug for us where the websocket Netty channel doesn't read from the channel, I created an issue for that and a PR with the fix: #7050

**Comment by asafalima** [comment]
Fixed missing copyright header

File: webserver/websocket/src/test/java/io/helidon/webserver/websocket/test/ConnectionClosingTest.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.webserver.websocket.test;
+
+import jakarta.websocket.*;
```
Comment (spericas) [inline webserver/websocket/src/test/java/io/helidon/webserver/websocket/test/ConnectionClosingTest.java:19]: Could you please expand this "*"? Thx.

Comment (asafalima) [reply webserver/websocket/src/test/java/io/helidon/webserver/websocket/test/ConnectionClosingTest.java]: Done! Let me know if there's something else I should fix.

**Review by romain-grecourt — APPROVED** [review]

**Pull request merged**